### DISCUSSION TITLE
License change

### DISCUSSION THREAD

**Discussion by jwmbeenakker** [MREYE-LUMC/ZOSPy#57] [General]
Currently, `ZOSPy` is licensed under [GPL-3.0 license](https://github.com/MREYE-LUMC/ZOSPy/discussions/new?category=general#GPL-3.0-1-ov-file). Within the open-source community this license has however become much debated for libraries such as `ZOSPy` and due its limitations is becoming less and less used. To enable a more easy (re)use of `ZOSPy`, we are therefore strongly considering to change to the more permissive [MIT license](https://choosealicense.com/licenses/mit/).

@crnh @LucVV @andibarg @Omnistic @noahrbn : As you have contributed to the code, we would rather not make this transition without your consent. Could you state below if you agree with this change, or if you have any objections against this step.

**Comment by crnh** [comment]
I am okay with this!

_Edit: only if this doesn't mean ANSYS can include ZOSPy in OpticStudio under a closed source license._

**Comment by LucVV** [comment]
So am I!

**Comment by noahrbn** [comment]
Thanks for asking. I have no issues with this, but as the creators I wonder if you are OK with commercial use of the package.

**Comment by andibarg** [comment]
I have nothing against it either. Thanks for asking.

But I am wondering the same as @noahrbn. Just out of interest: do I understand correctly that, for example, the MIT license would permit Zemax to include the ZOSPy code in an OpticStudio release and not be forced to make it open source? If they take over, we would probably not continue the development on github any longer. Are there any other scenarios this license change could entail?

**Reply by crnh** [reply]
This is probably the only scenario of commercial use we (as LUMC researchers) have problems with, but unfortunately also the most likely scenario. We indeed need to think about this. @jwmbeenakker @LucVV what do you think? GPL is not really suitable for a library, but maybe MIT is too permissive. What about LGPL ("GPL for libraries") or MPL (somewhat more permissive than LGPL)?

**Reply by jwmbeenakker** [reply]
Thanks for your replies. I find this a bit tricky, so it is good to have a discussion on this :). Some thoughts:
- I agree that the only commercial use of ZOSPy I could envision would be that OpticStudio would use it to improve their API, or examples. 
- However, ZOSPy requires OpticStudio to do anything useful (except having a clear documentation of your simulations). So although our code is freely accessible, it is not really "free" to use. Furthermore, we depend on the OpticStudio API, the real interesting part (the optics) is not opensource. This make ZOSPy a bit a "weird" open source project: the code is open, everybody can contribute, but you rely on closed software to use it.
- The question of Optic Studio incorporating ZOSPy is interesting, as this would basically mean that they would (finally) provide a proper/user-friendly interface to their software. I would be in favour of  such an interface but I don't foresee it in the near future since their API is platform independent. If it would be ZOSPy based, that would be nice, since that would mean that ZOSPy would still work with their new interface, and thus this repository will be valuable als people can contribute  (instead of the old [`PYZDDE`](https://github.com/xzos/PyZDDE) interface to which we also contributed, but became obsolete with the introduction of the .NET api).

But, personally, I don't have a preference for a specific license, except that I like to have a permissive license as this increases the change of it actually being used, also outside the research environment. 



Ps. Just to be clear: we have had no contact with OpticStudio regarding this proposed change, and we certainly have no plans of handing ZOSPy over to them. It is just that for another project we had discussions about licenses and this made me realise we might not have the most suitable license for ZOSPy.

**Reply by noahrbn** [reply]
As a complete newcomer to this repository and past user of `PYZDDE`, it is exceptional to me that the functionality of `ZOSPy` had to be an open-source project rather than something distributed by Ansys itself. It is odd that the burden of usability has been placed on the user community.

**Reply by crnh** [reply]
We discussed this internally, and the "risk" of Ansys taking over ZOSPy is not substantial in our opinion. Even if this happens, it is not likely to be a bad thing.

- ZOSPy will remain open source. Ansys can fork ZOSPy, but they can't take down this repository. If necessary, we can continue developing an open source library to control OpticStudio.
- [According to a post on the OpticStudio forum by Mark Nicholson](https://community.zemax.com/zos-api-12/zospy-v1-0-0-is-out-4079?postid=14344#post14344), Zemax did not intend to create an easy-to-use Python interface, but opted for the current "language-agnostic" dotnet interface instead.
- Integration of ZOSPy in OpticStudio will improve the user experience and would in the end improve our lives. I don't see how it would be beneficial for Ansys to include a closed source version; for ease of use, the Python library needs to be distributed using PyPI anyways. Furthermore, Ansys made other [Python code, including OpticStudio examples](https://github.com/ansys/optical-automation), available under an open source license. Lastly, given the speed at which the ZOS-API is developed, this will not happen.

**Comment by Omnistic** [comment]
Sorry for being late to the party, and thanks for asking my opinion. I'm a regular user of the MIT license and I am completely ok with this. Thanks all for the good work!

**Comment by jwmbeenakker** [comment]
Thank you all for your thoughts and support. We have just made the license change (#58 ).
Have a nice Christmas and looking forward to continue improving/expanding ZOSPy in the next year (#51)
